modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java

```diff
@@ -321,6 +326,35 @@ public void testGetCustomizedSubmitLabel() throws Exception {
 			submitLabel, ddmFormDisplayContext.getSubmitLabel());
 	}
 
+	@Test
+	public void testGetFormInstanceWithMissingSettings() throws Exception {
+		DDMFormDisplayContext ddmFormDisplayContext =
+			_createDDMFormDisplayContext();
+
+		DDMFormInstance ddmFormInstance = spy(new DDMFormInstanceImpl());
+
+		String expectedSettings = StringUtil.randomString();
+
+		ddmFormInstance.setSettings(expectedSettings);
+
+		when(
+			_ddmFormInstanceService.fetchFormInstance(Matchers.anyLong())
+		).thenReturn(
+			ddmFormInstance
+		);
+
+		when(
+			_ddmFormInstanceVersion.getSettings()
+		).thenReturn(
+			StringPool.BLANK
+		);
+
+		ddmFormDisplayContext.getFormInstance();
```

Comment:
Hey @natocesarrego, I couldn't understand this. Shouldn't it be `ddmFormInstance 
 = ddmFormDisplayContext.getFormInstance();`?

Reply:
Nice catch, @carolmariaabb! :smile: 

I didn't add it before because it wasn't really necessary since I was mocking the return of `_ddmFormInstanceService.fetchFormInstance`, even though I'm including your suggestion to make it clearer to the developers reading this code. :wink: 

Thanks, Carolzinha!